secp256k1-sys/vendor-libsecp.sh

```diff
@@ -1,42 +1,92 @@
-#!/bin/bash
+#!/usr/bin/env bash
 set -e
 
+# Set default variables
+: "${SECP_VENDOR_GIT_ROOT:=$(git rev-parse --show-toplevel)}"
```

Comment:
TIL, nice syntax.

Reply:
TBH I don't know what it means. I copied it from somebody smarter than me, and add the quotes around it because shellcheck claims there's some sort of globbing-related DoS otherwise.

Reply:
After spending a bunch of time reading about this:

* `${variable}` is familiar syntax which evaluates to the value of `variable`. But you can alternately do `${variable=xyz}` which, if `variable` is unset, will assign `xyz` to it and return that. `${variable:=xyz}` is the same but applies if `variable` is unset **or** "null",whatever that means
* So you can get this "assign if null" behavior but only in variable expansions. So if you just did `${variable:=xyz}` on a line by itself, bash would try to evaluate `xyz` (or whatever) and give you an error message.
* To avoid this, you pass to `:` which is a shell bulit-in which is synonymous with `true`. That is, it's a no-op command that eats its arguments. So you can get the side-effect of evaluating the arguments without having to deal with their pesky values.

Anyway somehow this has become a bash idiom. TIL.

Reply:
Cheers, I had worked out the first colon but not the second,cheers.